lib/spack/spack/solver/asp.py

```diff
@@ -1459,11 +1472,11 @@ def target_defaults(self, specs):
                 self.target_ranges(spec, None)
                 continue
 
-            if target not in compatible_targets:
-                compatible_targets.append(target)
+            if target not in candidate_targets:
+                candidate_targets.append(target)
```

Comment:
I think we need to add the ancestors of the target here as well. If I'm on an aarch64 system, and try to concretize `hdf5 target=broadwell`, then I should have `haswell` and `x86_64` available as allbacks for dependency nodes that can't build with broadwell.

Reply:
See ec1bb5f I think we'll need to revisit anyhow when adding proper support for cross-compilation.